src/main/java/org/spongepowered/api/GameRegistry.java

```diff
@@ -825,4 +834,62 @@
      */
     Optional<Translation> getTranslationById(String id);
 
+    /**
+     * Gets a new {@link WorldBuilder} for creating {@link World}s or
+     * {@link WorldCreationSettings}s.
+     * 
+     * @return A new builder
+     */
+    WorldBuilder getWorldBuilder();
+
+    /**
+     * Gets a new {@link WorldBuilder} for creating {@link World}s or
+     * {@link WorldCreationSettings}s, the builder is then seeded with the
+     * values from the given WorldCreationSettings object.
+     * 
+     * @param settings The seed settings
+     * @return A new seeded builder
+     */
+    WorldBuilder getWorldBuilder(WorldCreationSettings settings);
+
+    /**
+     * Gets a new {@link WorldBuilder} for creating {@link World}s or
+     * {@link WorldCreationSettings}s, the builder is then seeded with the
+     * values from the given WorldProperties object.
+     * 
+     * @param properties The seed properties
+     * @return A new seeded builder
+     */
+    WorldBuilder getWorldBuilder(WorldProperties properties);
+
+    /**
+     * Gets a {@link GeneratorType} by name.
+     * 
+     * @param name The requested name
+     * @return The generator type, if available
+     */
+    Optional<GeneratorType> getGeneratorType(String name);
+
+    /**
+     * Gets a collection of all available {@link GeneratorType}s.
+     * 
+     * @return All available world types
+     */
+    Collection<GeneratorType> getGeneratorTypes();
+    
+    /**
+     * Creates and registers a new {@link GeneratorType} with the given name and
+     * {@link ChunkGenerator}.
+     * 
+     * <p>The {@link DataContainer} with settings for the world generator is
+     * optional and an empty DataContainer will be created if it is not
+     * provided.</p>
+     * 
+     * @param name The name of the generator type
+     * @param generator The generator for the type to use
+     * @param settings Any settings that the generator uses, or null if no
+     *            settings are used
+     */
+    void registerGeneratorType(String name, ChunkGenerator generator, @Nullable DataContainer settings);
```

Comment:
Good points, I think my thinking at the time was that a generator is stateless, but thats not quite right and doesn't match the method in GeneratorType.